pkg/collector/corechecks/cluster/orchestrator/discovery/collector_discovery_crd.go

```diff
@@ -33,29 +35,25 @@ type DiscoveryCollector struct {
 }
 
 func NewDiscoveryCollectorForInventory() *DiscoveryCollector {
-	return &DiscoveryCollector{
+	dc := &DiscoveryCollector{
 		cache: discoveryCache{collectorForVersion: map[collectorVersion]struct{}{}},
 	}
-}
-
-func (d *DiscoveryCollector) VerifyForInventory(resource string, groupVersion string) (collectors.Collector, error) {
-	collector, err := d.DiscoverCRDResource(resource, groupVersion)
+	err := dc.fillCache()
 	if err != nil {
-		return nil, err
+		log.Error("Fail to init discovery collector", err)
 	}
-	return collector, nil
+	return dc
 }
-
```

Comment:
Any risk of us returning the dc if a fillCache fails?

Reply:
If fillCache fails, it means we can't find any available resources from the server. 
 
In this case, we won't find any collector here https://github.com/DataDog/datadog-agent/blob/2796f30a2c7d53f4faca24d42afb2a52e3fa1cfb/pkg/collector/corechecks/cluster/orchestrator/discovery/collector_discovery_crd.go#L113-L121

Therefore, our bundle won't be able to load any collector from the config https://github.com/DataDog/datadog-agent/blob/2796f30a2c7d53f4faca24d42afb2a52e3fa1cfb/pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go#L148-L153 , which I think is expected.